<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>SRST2 - Short Read Sequence Typing</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <script src="javascripts/scale.fix.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1 class="header">SRST2</h1>
        <p class="header">Short Read Sequence Typing for Bacterial Pathogens</p>

        <ul>
          <li><a class="buttons github" href="https://github.com/katholt/srst2">View On GitHub</a></li>
          <li class="download"><a class="buttons" href="https://github.com/katholt/srst2/archive/v0.1.4.zip">Download ZIP</a></li>
          <li class="download"><a class="buttons" href="https://github.com/katholt/srst2/archive/v0.1.4.tar.gz">Download TAR</a></li>
        </ul>

        <p class="header">This project is maintained by <a class="header name" href="mailto:drkatholt@gmail.com">katholt</a></p>


      </header>
      <section>
        <h1>
<a name="srst2" class="anchor" href="#srst2"><span class="octicon octicon-link"></span></a>SRST2</h1>

<h2>Short Read Sequence Typing for Bacterial Pathogens</h2>

<p>This program is designed to take Illumina sequence data, a MLST database and/or a database
of gene sequences (e.g. resistance genes, virulence genes, etc) and report the presence of
STs and/or reference genes.</p>

<h3>Dependencies:</h3>

<ul>
<li>python (v2.7.5), scipy, numpy</li>

<li>bowtie2 v2.1.0     <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">http://bowtie-bio.sourceforge.net/bowtie2/index.shtml</a></li>

<li>SAMtools v0.1.18   <a href="https://sourceforge.net/projects/samtools/files/samtools/0.1.18/">https://sourceforge.net/projects/samtools/files/samtools/0.1.18/</a> (NOTE 0.1.19 DOES NOT WORK)</li>
</ul>

<P>
<h3>Manuscript: <a href="http://genomemedicine.com/content/6/11/90">Inouye et al, Genome Medicine, 2014</a></h3>
Abstract: Rapid molecular typing of bacterial pathogens is critical for public health epidemiology, surveillance and infection control, yet routine use of whole genome sequencing (WGS) for these purposes poses significant challenges. Here we present SRST2, a read mapping-based tool for fast and accurate detection of genes, alleles and multi-locus sequence types (MLST) from WGS data. Using >900 genomes from common pathogens, we show SRST2 is highly accurate and outperforms assembly-based methods in terms of both gene detection and allele assignment. We include validation of SRST2 within a public health laboratory, and demonstrate its use for microbial genome surveillance in the hospital setting. In the face of rising threats of antimicrobial resistance and emerging virulence among bacterial pathogens, SRST2 represents a powerful tool for rapidly extracting clinically useful information from raw WGS data.
</P>

<h3>Start with <a href="https://github.com/katholt/srst2/blob/master/README.md">installation and documentation</a> and <a href="https://github.com/katholt/srst2/blob/master/example.txt">example/tutorial</a></h3>

<h3>Bugs and feature requests? <a href="https://github.com/katholt/srst2/issues">Please use github issue tracking</a></h3>

<h3>Want to hear about updates? <a href="https://groups.google.com/forum/#!forum/srst2">Subscribe to the SRST2 google group</a></h3>

<hr>
<h4>Updates in v0.1.4 (June 26, 2014)</h4>

<ul>
<li>No longer store sam and unsorted bam (can be retained via the --keep_interim_alignment flag)</li>

<li>Added options to specify a maximum number of mismatches to allow during mapping; this is specified separately for mlst and genes, so that it is possible to relax the stringency of gene detection in the same run as a high-accuracy MLST test.
Default value for both is 10 mismatches.
--mlst_max_mismatch
--gene_max_mismatch</li>

<li>The highest minor allele frequency (MAF) of variants encountered in the alignment is now calculated and reported for each allele (in the scores file) and also at the gene level and ST level, to facilitate checking for mixed/contaminated read sets. 

This value is in the range 0 -> 0.5; with e.g. 0 indicating no variation between reads at any aligned base (i.e. at all positions in the alignment, all aligned reads agree on the same base call; although this agreed base may be different from the reference); and 0.25 indicating there is at least one position in the alignment at which all reads do not agree, and the least common variant (either match or mismatch to the reference) is present in 25% of reads. This value is printed, for all alleles, to the scores file. Note this is different to the ‘LeastConfident’ information printed to scores, which presents the strongest evidence for mismatch *compared to the reference*, i.e. between 0 -> 1.

The highest such value for each gene/cluster/locus is reported in the fullgenes output table.

The highest such value across all MLST loci is reported in the mlst output table.

Note that all compiled reports will now include a maxMAF column; if you provide MLST or compiled reports from previous versions without this columns, the value “NC” will be inserted in the maxMAF column to indicate “not calculated”. This ensures the updated SRST2 (v0.1.4+) is backwards compatible with previous SRST2 outputs; do be aware though that the older versions of SRST2 (prior to v0.1.4) will not be forwards-compatible with output generated by more recent versions (v0.14 onwards).
</li>
<li>Added R code for plotting SRST2 output in R (plotSRST2data.R).
Instructions will be added to the read me.</li>

</ul>

      </section>
      <footer>
        <p><small>Hosted on <a href="https://pages.github.com">GitHub Pages</a> using the Dinky theme</small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
		          <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-44427590-1");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>

  </body>
</html>
